### ISSUE TITLE
x/pkgsite: create focus indicators to replace browser defaults

### ISSUE THREAD

**Issue by jamalc** [golang/go#42698]
The default style and behavior of focus indicators is set by browsers to make websites more accessible to keyboard only users and those with reduced vision. Their styles vary across browsers and sometimes clash with pkgsite's layout (#40759) and design (#42583). Focus visibility is a WCAG requirement so we can remove the default outlines but we'll need replacement indicators. 

When replaced by a box shadow `:focus {outline: transparent; }` should be used in place of `:focus {outline: none; }` because box shadows will not appear on some high contrast screens and transparent focus will.

/cc @Joanne881107 @fflewddur for product / UX input

**Comment by jamalc** [comment]
Done.
